src/main_menu.cpp

```diff
@@ -1072,7 +1071,10 @@ void main_menu::world_tab( const std::string &worldname )
 {
     // Create world
     if( sel2 == 0 ) {
-        world_generator->make_new_world();
+        WORLD *world = world_generator->make_new_world();
+        if( world->world_name < world_generator->all_worldnames()[last_world_pos] ) {
```

Comment:
This seems like you want equality instead of less than? I think that will also silence the warning. 
```suggestion
        if( world->world_name != world_generator->all_worldnames()[last_world_pos] ) {
```